addons/product/static/src/js/product_pricelist_report.js

```diff
@@ -1,317 +1,242 @@
+            html: markup(""),
+        });
+
+        onWillStart(async () => {
+            this.state.pricelists = await this.getPricelist()
+            this.state.selectedPricelist = pastState.selectedPricelist || this.state.pricelists[0];
+
+            this.renderHtml();
+        });
+
+        onRendered(() => {
+            this.env.config.setDisplayName(this.displayName);
+        });
+
+        /*
+        When following the link of a product and coming back we need to keep the 
+        precendent state:
+            - if the pricelist was being showed
+            - wich pricelist is selected at the moment
+            - which quantities
+            - the products (or product variants) for which the report was asked
+            - if the active model is product.product or product.template
+        */
+        useSetupAction({
+            getLocalState: () => {
+                return {
+                    displayPricelist: this.displayPricelist,
+                    selectedPricelist: this.selectedPricelist,
+                    quantities: this.quantities,
+                    active_ids: this.active_ids,
+                    active_model: this.active_model,
+                };
+            },
+        });
+    }
+
+    get displayName() {
+        return this.env._t("Pricelist Report");
+    }
+
+    get quantities() {
+        return this.state.quantities;
+    }
+
+    get html() {
+        return this.state.html;
+    }
+
+    get displayPricelist() {
+        return this.state.displayPricelist;
+    }
+
+    get pricelists() {
+        return this.state.pricelists;
+    }
+
+    get selectedPricelist() {
+        return this.state.selectedPricelist;
+    }
+
+    get activeIds() {
+        return this.state.activeIds;
+    }
+
+    get activeModel() {
+        return this.state.activeModel;
+    }
+
+    get getReportParams() {
+        return {
+            active_model: this.activeModel || 'product.template',
+            active_ids: this.state.activeIds || [],
+            is_visible_title: this.displayPricelist || '',
+            pricelist_id: this.selectedPricelist.id || '',
+            quantities: this.quantities || [1],
+        };
+    }
+
+    // orm calls
+
+    async renderHtml() {
+        let html = await this.orm.call(
+            "report.product.report_pricelist", "get_html", [], {data: this.getReportParams}
+        );
+        this.state.html = markup(html);
+    }
+
+    getPricelist() {
+        return this.orm.searchRead("product.pricelist", [], ["id", "name"]);
+    }
+
+    // events
+
+    onClickPrint() {
+        this.action.doAction({
+            type: 'ir.actions.report',
+            report_type: 'qweb-pdf',
+            report_name: 'product.report_pricelist',
+            report_file: 'product.report_pricelist',
+            data: this.getReportParams,
+        });
+    }
+
+    toggleDisplayPricelist() {
```

Comment:
```suggestion
    onToggleDisplayPricelist() {
```

let's harmonize those handler names